src/Analyzers/CSharp/Analyzers/UseNullPropagation/CSharpUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -11,95 +11,92 @@
 using Microsoft.CodeAnalysis.LanguageService;
 using Microsoft.CodeAnalysis.UseNullPropagation;
 
-namespace Microsoft.CodeAnalysis.CSharp.UseNullPropagation
+namespace Microsoft.CodeAnalysis.CSharp.UseNullPropagation;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+internal sealed class CSharpUseNullPropagationDiagnosticAnalyzer :
+    AbstractUseNullPropagationDiagnosticAnalyzer<
+        SyntaxKind,
+        ExpressionSyntax,
+        StatementSyntax,
+        ConditionalExpressionSyntax,
+        BinaryExpressionSyntax,
+        InvocationExpressionSyntax,
+        ConditionalAccessExpressionSyntax,
+        ElementAccessExpressionSyntax,
+        MemberAccessExpressionSyntax,
+        IfStatementSyntax,
+        ExpressionStatementSyntax>
 {
-    [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    internal class CSharpUseNullPropagationDiagnosticAnalyzer :
-        AbstractUseNullPropagationDiagnosticAnalyzer<
-            SyntaxKind,
-            ExpressionSyntax,
-            StatementSyntax,
-            ConditionalExpressionSyntax,
-            BinaryExpressionSyntax,
-            InvocationExpressionSyntax,
-            ConditionalAccessExpressionSyntax,
-            ElementAccessExpressionSyntax,
-            MemberAccessExpressionSyntax,
-            IfStatementSyntax,
-            ExpressionStatementSyntax>
-    {
-        protected override SyntaxKind IfStatementSyntaxKind => SyntaxKind.IfStatement;
+    protected override SyntaxKind IfStatementSyntaxKind
+        => SyntaxKind.IfStatement;
 
-        protected override bool ShouldAnalyze(Compilation compilation)
-            => compilation.LanguageVersion() >= LanguageVersion.CSharp6;
+    protected override bool ShouldAnalyze(Compilation compilation)
+        => compilation.LanguageVersion() >= LanguageVersion.CSharp6;
 
-        protected override ISyntaxFacts GetSyntaxFacts()
-            => CSharpSyntaxFacts.Instance;
+    protected override ISemanticFacts SemanticFacts
+        => CSharpSemanticFacts.Instance;
+
+    protected override bool TryAnalyzePatternCondition(
+        ISyntaxFacts syntaxFacts, ExpressionSyntax conditionNode,
+        [NotNullWhen(true)] out ExpressionSyntax? conditionPartToCheck, out bool isEquals)
+    {
+        conditionPartToCheck = null;
+        isEquals = true;
 
-        protected override bool IsInExpressionTree(SemanticModel semanticModel, SyntaxNode node, INamedTypeSymbol? expressionTypeOpt, CancellationToken cancellationToken)
-            => node.IsInExpressionTree(semanticModel, expressionTypeOpt, cancellationToken);
```

Comment:
no longer need this override.  IsInExpressionTree is available on ISemanticFacts.  So this moved into the base class.